static/js/src/advantage/credentials/components/CredCheckout/CredCheckout.tsx

```diff
@@ -0,0 +1,102 @@
+import React from "react";
+import { QueryClient, QueryClientProvider } from "react-query";
+import { ReactQueryDevtools } from "react-query/devtools";
+import * as Sentry from "@sentry/react";
+import { Integrations } from "@sentry/tracing";
+import { Elements } from "@stripe/react-stripe-js";
+import { loadStripe } from "@stripe/stripe-js";
+import { UserSubscriptionMarketplace } from "advantage/api/enum";
+import Checkout from "advantage/subscribe/checkout/components/Checkout";
+import {
+  Action,
+  LoginSession,
+  Product,
+} from "advantage/subscribe/checkout/utils/types";
+
+const oneHour = 1000 * 60 * 60;
+const queryClient = new QueryClient({
+  defaultOptions: {
+    queries: {
+      refetchOnWindowFocus: false,
+      refetchOnMount: false,
+      refetchOnReconnect: false,
+      staleTime: oneHour,
+      retryOnMount: false,
+    },
+  },
+});
+
+Sentry.init({
+  dsn: "https://0293bb7fc3104e56bafd2422e155790c@sentry.is.canonical.com//13",
+  integrations: [
+    new Integrations.BrowserTracing({
+      tracingOrigins: ["ubuntu.com"],
+    }),
+  ],
+  allowUrls: ["ubuntu.com"],
+  tracesSampleRate: 1.0,
+});
+
+declare global {
+  interface Window {
+    accountId?: string;
+    stripePublishableKey?: string;
+    loginSession?: LoginSession;
+    marketplace: UserSubscriptionMarketplace;
+    canTrial?: boolean;
+    currentPaymentId: string;
+    invoiceId: string;
+    captcha: string | null;
+    previousPurchaseIds?: { monthly: string; yearly: string };
+    GAFriendlyProduct?: {
+      id: string;
+      name: string;
+      price: number;
+      quantity: number;
+    };
+  }
+}
+const checkoutData = localStorage.getItem("shop-checkout-data") || "{}";
+const parsedCheckoutData = JSON.parse(checkoutData);
+const stripePromise = loadStripe(window.stripePublishableKey || "");
+const product: Product = parsedCheckoutData?.product;
+const quantity: number = parsedCheckoutData?.quantity;
+const action: Action = parsedCheckoutData?.action;
+console.log(product, quantity, action);
+window.previousPurchaseIds = {
+  monthly: "",
+  yearly: "",
+};
+window.canTrial = undefined;
+window.currentPaymentId = "";
+window.accountId = "";
+window.captcha = null;
+
+window.marketplace = product?.marketplace;
+window.GAFriendlyProduct = {
+  id: product?.id,
+  name: product?.name,
+  price: (product?.price?.value ?? 0) / 100,
+  quantity: quantity,
+};
+
+const CredCheckout = () => {
```

Comment:
Why do you duplicate the whole Checkout app? You should not do that.

Your credential shop should set the product data in the localstorage and reuse the checkout app. You needed a custom redirect and you added an extra parameter for that. That's all correct.
When you add that parameter you have to make sure on all other places the `Checkout` component is being used, those would still work. 

What else is the `Checkout` missing that requires you to duplicate this whole logic?

Reply:
I am not duplicating the Checkout here, I am duplicating the [static/js/src/advantage/subscribe/checkout/app.tsx.](https://github.com/canonical/ubuntu.com/blob/9f5ffd6033b4cdb768b39f0520f2d33c460a00bd/static/js/src/advantage/subscribe/checkout/app.tsx) app.tsx is not an exported component so it won't work outside your application, hence I have to duplicate it. The thing is that you are initializing the product in the base app itself but I have to do it in a separate checkout component which then references your original checkout component. I think the component could be renamed to something like CheckoutWrapper to avoid confusion.